source/dagger/gameplay/team_game/team_game_collisions.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include "core/system.h"
+#include "core/core.h"
+#include <array>
+
+using namespace dagger;
+
+namespace team_game
+{
+    enum class CollisionSide { NONE, RIGHT, LEFT, TOP, BOTTOM };
+
+    enum class PlatformerCollisionID { PLAYER, TERRAIN, ENEMY };
+
+    enum class MovementState { IMMOBILE, UNSTOPPABLE, MOVEABLE };
+
+    struct CollisionInfo
+    {
+        Bool hasCollided{ false };
+
+        CollisionSide collisionSide = CollisionSide::NONE, collisionSideOther = CollisionSide::NONE;
+    };
+
+    struct Collider
+    {
+        PlatformerCollisionID entityType = PlatformerCollisionID::PLAYER;//this is a placeholder and should be changed to not be automatically player
+        MovementState state = MovementState::MOVEABLE;
+
+        Vector2 size;
+        Vector2 pivot{ -0.5f, -0.5f };
+
+        Bool hasGravity{ false };
+
+        Bool canGoRight{ true };
+        Bool canGoLeft{ true };
+        Bool canGoUp{ true };
+        Bool canGoDown{ true };
+
+        std::vector<entt::entity> listOfEntities;//holds list of entities that this object collided with during this frame
+        std::vector<CollisionSide> listOfCollisionSides;//holds the corresponding list of sides of collisions
+
+        // player, terrain, enemies, traps, collectables, weapons
+        // For player it would be: false, true, true, true, ???, false
+        StaticArray<Bool, 6> collidesWith{ true, true, true, true, true, false };
+
+        CollisionInfo GetCollisionInfo(Vector3& pos_, const Collider& other_, Vector3& posOther_);
+
+    };
+
+    class PlatformerCollisionSystem : public System
```

Comment:
Might want to rename this and PlatformerCollisionID (line 12) to just CollisionSystem and CollisionID

Reply:
Will do, its a relic from the old system based on the platformer example

Reply:
Renamed